internal/provider/search_attribute_resource.go

```diff
@@ -0,0 +1,292 @@
+	return &SearchAttributeResource{}
+}
+
+// SearchAttributeResource - a Temporal search attribute resource implementation.
+type SearchAttributeResource struct {
+	client grpc.ClientConnInterface
+}
+
+// SearchAttributeResourceModel defines the data schema for a Temporal search attribute resource.
+type SearchAttributeResourceModel struct {
+	Name      types.String `tfsdk:"name"`
+	Type      types.String `tfsdk:"type"`
+	Namespace types.String `tfsdk:"namespace"`
+}
+
+// Metadata sets the metadata for the  search attribute resource, specifically the type name.
+func (r *SearchAttributeResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_search_attribute"
+}
+
+// Schema returns the schema for the Temporal search attribute resource.
+func (r *SearchAttributeResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Temporal Search Attribute resource",
+
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Search Attribute Name",
+				Required:            true,
+			},
+			"type": schema.StringAttribute{
+				MarkdownDescription: "Search Attribute Indexed Value Type, which defines the type of data stored in the attribute",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("Unspecified", "Text", "Keyword", "Int", "Double", "Bool", "Datetime", "KeywordList"), // Ensure only valid types are used
+				},
+			},
+			"namespace": schema.StringAttribute{
+				MarkdownDescription: "Namespace with which the Search Attribute is associated",
+				Optional:            true,
+				Default:             stringdefault.StaticString("default"),
+				Computed:            true,
+			},
+		},
+	}
+}
+
+// Configure sets up the search attribute resource configuration.
+func (r *SearchAttributeResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	tflog.Info(ctx, "Configuring Temporal Search Attribute Resource")
+
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(grpc.ClientConnInterface)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected grpc.ClientConnInterface, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+	tflog.Info(ctx, "Configured Temporal Search Attribute client", map[string]any{"success": true})
+}
+
+// Create is responsible for creating a new search attribute in Temporal.
+func (r *SearchAttributeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
```

Comment:
Do we check anywhere if the new attribute the user tries to create is already present as part of default system search attributes? 
I guess the behavior should be - users are not allowed to create attributes with same name/type as what exists as part of default system attributes right?

Reply:
I believe ideally the API doesn't allow to create a search attribute if it already exists, including the default system search attributes. As per the API documentation [here](https://pkg.go.dev/go.temporal.io/api@v1.30.1/operatorservice/v1#OperatorServiceClient.AddSearchAttributes). But the API is not throwing the error as expected, and that's why @amibaid is going to start a support request with the temporal team to understand why it is not working as expected. Also CLI is throwing the error, so the behavior is not consistent across CLI and API. If the API doesn't support it, then @amibaid is going to add the logic to error out in case if the attribute already exists.

Reply:
The API already correctly throws an error if the user tries to create an attribute with the same name as a system attribute.  As for the custom attributes, I have opened a support thread to investigate why API is not behaving as expected

Reply:
[Temporal support](https://temporalio.slack.com/archives/CTDTU3J4T/p1721258119875649?thread_ts=1721255485.197359&cid=CTDTU3J4T) has indicated API docs may be out of date, so I have added error handling for this case in the code